### PR TITLE
Remove unneeded Auth Extension files

### PR DESCRIPTION
**Pull request by damemi**
Now that the GCP auth extension is available in [collector-contrib](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/cec2153af5221eff3b2cec0172f4c689669f36f8/extension/googleclientauthextension) we don't need some of this code:

* Removes `testdata/config.yaml` and `config_test.go` (duplicated [upstream[(https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/extension/googleclientauthextension/config_test.go))
* Removes README steps about building into a custom collector
* Removes `internal/metadata/status.go` as it is [duplicated upstream](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/extension/googleclientauthextension/internal/metadata/generated_status.go)

### CODE REVIEW

File: extension/googleclientauthextension/factory_test.go [hunk 1]
```diff
@@ -19,32 +19,43 @@ import (
...
-	ext, err := NewFactory().CreateExtension(context.Background(), extension.CreateSettings{}, CreateDefaultConfig())
+	ext, err := newFactory().CreateExtension(context.Background(), extension.CreateSettings{}, CreateDefaultConfig())
 	assert.NotNil(t, ext)
 	assert.NoError(t, err)
 }
 
 func TestStart(t *testing.T) {
 	t.Setenv("GOOGLE_APPLICATION_CREDENTIALS", "testdata/fake_creds.json")
-	ext, err := NewFactory().CreateExtension(context.Background(), extension.CreateSettings{}, CreateDefaultConfig())
+	ext, err := newFactory().CreateExtension(context.Background(), extension.CreateSettings{}, CreateDefaultConfig())
```
Comment (dashpole) [inline extension/googleclientauthextension/factory_test.go:58]:
```suggestion
	ext, err := CreateExtension(context.Background(), extension.CreateSettings{}, CreateDefaultConfig())
```

File: extension/googleclientauthextension/factory_test.go [hunk 2]
```diff
@@ -53,7 +64,7 @@ func TestStart(t *testing.T) {
 
 func TestStart_WithError(t *testing.T) {
 	t.Setenv("GOOGLE_APPLICATION_CREDENTIALS", "testdata/foo.json")
-	ext, err := NewFactory().CreateExtension(context.Background(), extension.CreateSettings{}, CreateDefaultConfig())
+	ext, err := newFactory().CreateExtension(context.Background(), extension.CreateSettings{}, CreateDefaultConfig())
```
Comment (dashpole) [inline extension/googleclientauthextension/factory_test.go:67]:
```suggestion
	ext, err := CreateExtension(context.Background(), extension.CreateSettings{}, CreateDefaultConfig())
```

File: extension/googleclientauthextension/factory_test.go [hunk 3]
```diff
@@ -19,32 +19,43 @@ import (
 	"testing"
 
 	"github.com/stretchr/testify/assert"
+	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/component/componenttest"
 	"go.opentelemetry.io/collector/extension"
 )
 
+// newFactory creates a factory for the GCP Auth extension.
+func newFactory() extension.Factory {
```
Comment (dashpole) [inline extension/googleclientauthextension/factory_test.go:28]: I don't think this is necessary.  You should just call the public API (CreateDefaultConfig and CreateExtension) as needed to test them.

Comment (damemi) [reply extension/googleclientauthextension/factory_test.go]: Yup you're totally right, updated all suggestions above

**Review by dashpole — APPROVED** [review]

**Pull request merged**